### PR TITLE
[WIP] Add transfers flow

### PR DESCRIPTION
**Pull request by EugeneIOs**

### CODE REVIEW

File: Sources/VGSCheckoutSDK/Core/Checkout/CheckoutDelegate/VGSCheckoutDelegate.swift [hunk 1]
```diff
@@ -10,6 +10,18 @@ public protocol VGSCheckoutDelegate: AnyObject {
...
 	func checkoutDidCancel()
+
+	/// Tells the delegate that save card succeeded and fin instrument created.
+	/// - Parameter data: `Data?` object, holds response data with created financial instrument for saved card.
+	/// - Parameter response: `URLResponse?` object, holds URL response object with created financial instrument for saved card.
+	func saveCardDidSuccess(with data: Data?, response: URLResponse?)
+}
+
+/// Defeault protocol implementation.
+public extension VGSCheckoutDelegate {
```
Comment (dmytrokhl) [inline Sources/VGSCheckoutSDK/Core/Checkout/CheckoutDelegate/VGSCheckoutDelegate.swift:23]: should we comment this code for now?

File: Sources/VGSCheckoutSDK/Core/Checkout/UseCases/Payment/ViewControllers/PaymentOptions/ViewModel/VGSPaymentOptionsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+						if savedCardIndex != index {
+							print("unmard card!")
+							previousCard.isSelected = false
+						}
+					case .newCard:
+						continue
+					}
+				}
+			}
+			print("current new savedCard.id: \(savedCard.id)")
```
[the remark this answers is not in the document]
Comment (EugeneIOs) [reply Sources/VGSCheckoutSDK/Core/Checkout/UseCases/Payment/ViewControllers/PaymentOptions/ViewModel/VGSPaymentOptionsViewModel.swift]: Commented for now.
Done.

File: Sources/VGSCheckoutSDK/Core/Checkout/UseCases/Payment/Models/PaymentOptions/SavedCardModel/VGSSavedCardModel.swift [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	}
+}
+
+// no:doc
+internal extension Array where Element == VGSSavedCardModel {
+	/// Reorders array of saved cards by fin instrument ids.
+	/// - Parameter cardIds: `[String]` object, an array of fin instrument ids.
+	/// - Returns: `[VGSSavedCardModel]` object, an array of reordered saved cards.
+	func reorderByIds(_ cardIds: [String]) -> [VGSSavedCardModel] {
+		var orderedArray: [VGSSavedCardModel] = []
```
[the remark this answers is not in the document]
Comment (EugeneIOs) [reply Sources/VGSCheckoutSDK/Core/Checkout/UseCases/Payment/Models/PaymentOptions/SavedCardModel/VGSSavedCardModel.swift:105]: This sorting is not for ascending sort.

File: Sources/VGSCheckoutSDK/Core/Checkout/CheckoutDelegate/VGSCheckoutDelegate.swift (hunk 1 above)
Comment (EugeneIOs) [reply Sources/VGSCheckoutSDK/Core/Checkout/CheckoutDelegate/VGSCheckoutDelegate.swift:27]: We can do this when API will be finalized.

**Review by dmytrokhl — APPROVED** [review]

**Pull request merged**